pycalphad/model.py

```diff
@@ -137,15 +137,22 @@ def __init__(self, dbe, comps, phase_name, parameters=None):
 
         # Verify that this phase is still possible to build
         for sublattice in phase.constituents:
-            if len(set(sublattice).intersection(self.components)) == 0:
+            sublattice_comps = set(sublattice).intersection(self.components)
+            if len(sublattice_comps) == 0:
                 # None of the components in a sublattice are active
                 # We cannot build a model of this phase
                 raise DofError(
                     '{0}: Sublattice {1} of {2} has no components in {3}' \
                     .format(self.phase_name, sublattice,
                             phase.constituents,
                             self.components))
-            self.constituents.append(set(sublattice).intersection(self.components))
+            if sum(set(map(lambda s : getattr(s, 'number_of_atoms'),sublattice_comps))) == 0:
```

Comment:
Thanks for submitting this PR! 

I think this line may be causing the tests to fail because *any* sublattice that has only VA will raise an error, but we only want to raise an error here if *all* sublattices have only `VA`. 

For example, given a phase with sublattices `(('FE', 'VA'), ('VA'))`, the phase should built if both `FE` and `VA` are specified (i.e. it's ok if the last sublattice is `VA` only, as long as the first sublattice has `FE` occupied).

It might be helpful to write a unit test that exercises some different conditions.

Reply:
You're welcome! Its a pleasure to try contributing on pycalphad.

I see, as you said, my previous approach works detecting if the phase has any pure vancacy sublattice. I fixed this storing the `number_of_atoms` of each sublattice in a set called `is_pure_VA` and checking if the sum of `is_pure_VA` is equal to zero after get the `number_of_atoms` in each sublattice. Now i think that it will work, I tested building some `Models` used in the tests of TravisCI and it worked fine